### PR TITLE
Mongo driver 4.1.0

### PR DESCRIPTION
**Pull request by sebaciv**
Bump mongo driver to 4.1.0 and remove deprecated usages

### CODE REVIEW

File: commons-mongo/src/main/scala/com/avsystem/commons/mongo/async/GenCodecCollection.scala [hunk 1]
```diff
@@ -3,14 +3,15 @@ package mongo.async
 
 import com.avsystem.commons.mongo.core.GenCodecRegistry
 import com.avsystem.commons.serialization.GenCodec
-import com.github.ghik.silencer.silent
-import com.mongodb.async.client.{MongoCollection, MongoDatabase}
+import org.mongodb.scala.{MongoCollection, MongoDatabase}
 
 object GenCodecCollection {
```
Comment (ddworak) [inline commons-mongo/src/main/scala/com/avsystem/commons/mongo/async/GenCodecCollection.scala:8]: Does it differ from `com.avsystem.commons.mongo.scala.GenCodecCollection`? I guess we should remove this one now.

Comment (sebaciv) [reply commons-mongo/src/main/scala/com/avsystem/commons/mongo/async/GenCodecCollection.scala]: Okay, deleted. I also moved to `MongoObservableExtensions` to `scala` package

Comment (ddworak) [reply commons-mongo/src/main/scala/com/avsystem/commons/mongo/async/GenCodecCollection.scala]: Let's call it MongoScalaObservableExtensions as they're no longer generic to Mongo (and the Mongo ones have different interface based on reactive streams)

**Review by ddworak — APPROVED** [review]

**Comment by ddworak** [comment]
Please fix tests as well ;)

**Pull request merged**